core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -4431,6 +4431,14 @@ private void validateModality(SqlNode query) {
       default:
         break;
       }
+    } else if (query.getKind() == SqlKind.WITH) {
+    // The modality of With clause depends on its body
+    // For example:
+    // SQL: WITH STREMTABLE AS SELECT STREAM * FROM KAFKA.MOCKTABLE SELET * FROM STREMTABLE
```

Comment:
is this supposed to be STREAMTABLE?
SELECT is also mistyped

Reply:
Updated.